File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpSignTextReadData.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RMcpSignTextReadData(
        RMcpBlockPosData pos,
        boolean waxed,
        Side front,
        Side back
) {
    public record Side(
            String color,
            boolean glowing,
            List<String> lines,
            boolean hasText
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpSignTextReadData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpSignTextReadData(
    val pos: RBlockPos,
    val waxed: Boolean,
    val front: Side,
    val back: Side?
) {
    @JvmRecord
    data class Side(
        val color: String,
        val glowing: Boolean,
        val lines: List<String>,
        val hasText: Boolean
    )
}
